api/src/main/java/org/pmiops/workbench/db/dao/BillingProjectGarbageCollectionDao.java

```diff
@@ -0,0 +1,26 @@
+package org.pmiops.workbench.db.dao;
+
+import java.util.List;
+import org.pmiops.workbench.db.model.BillingProjectGarbageCollection;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface BillingProjectGarbageCollectionDao
+    extends CrudRepository<BillingProjectGarbageCollection, Long> {
+
+  // get Billing Projects which are ASSIGNED to Workspaces which have been DELETED and have NEW
+  // Billing Migration Status.  These are ready to be garbage-collected
+  @Query(
+      "SELECT p.fireCloudProjectName "
+          + "FROM BillingProjectBufferEntry p "
+          + "JOIN Workspace w "
+          + "ON w.workspaceNamespace = p.fireCloudProjectName "
+          + "AND p.status = 4 " // BillingProjectBufferStatus.ASSIGNED
```

Comment:
Is it common for us to use magic enum numbers in queries like this?

Reply:
Dunno - but yeah I should do this without magic